jenkins/spark-nightly-build.sh

```diff
@@ -25,6 +25,7 @@ mvn -U -B -Pinclude-databricks,snapshot-shims clean deploy $MVN_URM_MIRROR -Dmav
 # Run unit tests against other spark versions
 mvn -U -B -Pspark301tests,snapshot-shims test $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR
 mvn -U -B -Pspark302tests,snapshot-shims test $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR
+mvn -U -B -Pspark303tests,snapshot-shims test $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR
```

Comment:
Do we want the pre-commit build to start testing 303 too?

Reply:
Yes, I think that's a good thing.  I noticed we're not testing against spark302 in premerge, shall I add that while I'm at it?

Reply:
I think so. We really need to get to the point where we are dropping support for bugfix releases, but until then we need to test what we support. And it should only add 7 to 10 mins per version to the runtime of the premerge build.  Not ideal but acceptable for now.